src/include/duckdb/execution/index/art/node.hpp

```diff
@@ -38,6 +39,64 @@ enum class GateStatus : uint8_t {
 class ART;
 class Prefix;
 class ARTKey;
+class FixedSizeAllocator;
+
+//! State for TransformToDeprecated operations.
+//! Makes it explicit whether transformation is needed and contains the deprecated allocator.
+struct TransformToDeprecatedState {
```

Comment:
Let's turn this into a `class` so that we can have `public` and `private` parts.